Mage.Sets/src/mage/cards/e/EmpoweredAuogenerator.java

```diff
@@ -0,0 +1,46 @@
+package mage.cards.e;
+
+import mage.Mana;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldTappedAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.dynamicvalue.common.CountersSourceCount;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;;
+import mage.abilities.effects.mana.DynamicManaEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+
+import java.util.UUID;
+
+/**
+ * @author jmharmon
+ */
+
+public final class EmpoweredAutogenerator extends CardImpl {
+
+    public EmpoweredAutogenerator(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{4}");
+
+        // Empowered Autogenerator enters the battlefield tapped.
+        this.addAbility(new EntersBattlefieldTappedAbility());
+
+        // {T}: Put a charge counter on Empowered Autogenerator. Add X mana of any one color, where X is the number of charge counters on Empowered Autogenerator.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new AddCountersSourceEffect(CounterType.CHARGE.createInstance(), true), new TapSourceCost());
```

Comment:
This needs to be a mana ability, you should probably implement it as a `ManaEffect` in a `SimpleManaAbility`

Reply:
I'm not sure if I'll get this one. The closest cards I'm finding are Astral Cornucopia and Nykthos, but they're quite the same. The card reads like a `DynamicManaAbility`, it and the `SimpleActivatedAbility` together isn't working.

Reply:
It's a bit of a weird situation, which is why I hadn't gotten to it yet.